### DISCUSSION TITLE
Validation errors not set with custom input component

### DISCUSSION THREAD

**Discussion by slecorvaisier** [react-hook-form/react-hook-form#2951] [Q&A] [answered]
Hi there,

following the [documentation about UI lib](https://react-hook-form.com/get-started#IntegratingwithUIlibraries), the third option seems to fit the best my organization's project. While the form itself is working, I can't get any errors associated with the rules I specified.

I tried 2 different sandboxes to showcase it, I probably miss something but can't find what. Can anyone help me solving this issue while not using Controllers and refs (so focusing on "onChange" events as showcased in the doc's option 3)?

Thank you in advance!

CSB1: Reduced example from documentation:
https://codesandbox.io/s/react-hook-form-custom-input-forked-1nlpl?file=/src/index.js

CSB2: Example with simple custom Input:
https://codesandbox.io/s/react-hook-form-custom-input-forked-xw1od?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
https://codesandbox.io/s/react-hook-form-custom-input-forked-s76x1?file=/src/index.js

**Reply by slecorvaisier** [reply]
Even after several hours spent on the examples and the docs, I never noticed `valudationRule` was passed as a second parameter...
Thank you @bluebill1049 for your help.

**Reply by bluebill1049** [reply]
all good, mate. if you are using TS, it would have given you type.
